examples/vault-consul-ami/auth/sign-request.py

```diff
@@ -15,18 +15,28 @@
 # the response from GetCallerIdentity, which tells who is trying to authenticate
 # ------------------------------------------------------------------------------
 
-import botocore.session
-from botocore.awsrequest import create_request_object
-import json
 import base64
+import json
 import sys
 
-def headers_to_go_style(headers):
+import botocore.session
+
+
+def decode_bytes_from_dict_values(dict_, to_go_style=False):
     retval = {}
-    for k, v in headers.iteritems():
-        retval[k] = [v]
+    for k, v in dict_.items():
+        try:
+            value = v.decode()
+        except AttributeError:
+            value = v
+
+        if to_go_style:
+            value = [value]
+
+        retval[k] = value
     return retval
```

Comment:
I think using a single function for these two purposes is confusing. I would probably convert these function calls to inline dict comprehensions (which are 2 and 3 compatible) that do the things we want for each case.